### DISCUSSION TITLE
M600??

### DISCUSSION THREAD

**Discussion by SilverOfDragon** [mriscoc/Ender3V2S1#987] [Q&A]
i have a print i will make, but i need to know, can the firmware understand M600 stop command??

**Comment by mriscoc** [comment]
Yes, but [M600](https://marlinfw.org/docs/gcode/M600.html) is a filament change G-code

**Reply by fr3nch13** [reply]
Hey @mriscoc 

I'm currently running [Ender3V2-422-BLTUBL-MPC-20230809.bin](https://github.com/mriscoc/Ender3V2S1/releases/download/20230805/Ender3V2-422-BLTUBL-MPC-20230809.bin) on 2 printers, and [Ender3V2-427-BLTUBL-MPC-20230809.bin](https://github.com/mriscoc/Ender3V2S1/releases/download/20230805/Ender3V2-427-BLTUBL-MPC-20230809.bin) on 1. 

I've created a code file that includes the M600 after layer 2, but none of the 3 printers seem to recognize this G code. Is there some setting in the post-compiled firmware that I need to change? 

I'm also worried that if the M600 isn't being recognized, sill the filament runout sensors I have installed, still work?

I see the option to pause, and I've tried it, it works as expected, and even allows the purge option.

Any help would be greatly appreciated!

**Reply by mriscoc** [reply]
Verify if you write the M600 in upper case.

**Reply by fr3nch13** [reply]
It is indeed upper-case. 
Here is the code file that is generated from Cura 5.4
https://gist.github.com/fr3nch13/6e22c1539388dadd5fcc4f0e87b6435d

Specifically [Line 191](https://gist.github.com/fr3nch13/6e22c1539388dadd5fcc4f0e87b6435d#file-ce3e3v2_xyz_calibration_pla_filimentchange-gcode-L191)

**Reply by fr3nch13** [reply]
I tried to print it again, and verified the M600 was present, and the firmware still seems to ignore it.

```
;BEGIN FilamentChange plugin
M600
;END FilamentChange plugin
```

**Comment by fr3nch13** [comment]
Should it be in uppercase?On Aug 25, 2023, at 11:54 AM, Miguel Risco-Castillo ***@***.***> wrote:﻿
Verify if you write the M600 in upper case.

**Reply by mriscoc** [reply]
Yes, G-codes in upper case should work. M600 is working with both firmware builds. In the tune menu there is also the option for change filament.

**Reply by fr3nch13** [reply]
I put the gcode in a gist. Is there possibly something that Cure 5.4 is putting in it, that would disable M600?

**Reply by mriscoc** [reply]
Send the M600 via terminal (Pronterface, Octoprint) and check if the printer responds with any error.

**Reply by fr3nch13** [reply]
So I believe I may have figured it out. 
In Cura, the Post-processing script for filament change has an option called `Use Firmware Configuration`.  With it checked, the resulting lines in the g-code looks like below, and the firmware seems to ignore it:
```
;BEGIN FilamentChange plugin
M600
;END FilamentChange plugin
```

Without it checked, the resulting g-code looks like below, and the firmware actually responds:
```
;BEGIN FilamentChange plugin
M600 E30.00 U300.00 X0.00 Y0.00
;END FilamentChange plugin
```

The difference being the coordinates for where the print head should go to get its filament changed, which the checked option doesn't have.

**Testing M600 in the terminal.**
With the printer not printing/in standby.
In Octoprint, entering the `M600` code, just by itself, in the terminal, shows the firmware responding with below:
```
Send: M600
Recv: //action:paused
Printer signalled that it paused, switching state...
Recv: //action:prompt_end
Recv: //action:prompt_begin Pause
Recv: //action:prompt_button Dismiss
Recv: //action:prompt_show
Send: M876 S0
Recv: echo:busy: processing
Printer seems to support the busy protocol, will adjust timeouts and set busy interval accordingly
[...]
Recv: echo:busy: processing
[...]
Recv: echo:busy: processing
[...]
Recv: X:0.00 Y:0.00 Z:5.00 E:0.00 Count X:0 Y:0 Z:2000
Recv: echo:busy: processing
[...]
Recv: echo:busy: processing
[...]
Recv: echo:busy: processing
[...]
Recv: echo:busy: processing
[...]
Recv: X:240.00 Y:221.00 Z:20.00 E:0.00 Count X:19200 Y:17680 Z:8000
```
Which did move the print head. But within the g-code file, it still gets ignored.

I'll test on an active print and see if it works.

(I put in a bunch of SEO words in this post to hopefully help others googling.)
